src/librustdoc/html/render/mod.rs

```diff
@@ -561,10 +561,21 @@ fn document_short(
     }
 }
 
-fn document_full(w: &mut Buffer, item: &clean::Item, cx: &Context<'_>) {
+fn document_full(w: &mut Buffer, item: &clean::Item, cx: &Context<'_>, is_collapsible: bool) {
```

Comment:
nit: Could you use an enum instead? `document_full(w, item, cx, true)` doesn't tell me what the boolean means and could be confusing.

```rust
enum Collapsible { True, False }

// Example usage
document_full(w, item, cx, Collapsible::True);
```

Or, if you want to be able to use `if`s, you could use a wrapper struct. This might be a better approach.

```rust
struct Collapsible { is: bool }

// Example call
document_full(w, item, cx, Collapsible { is: true });

// Example check
if collapsible.is { /* ... */ }
```

I suggested naming the field `is` so that it's clear in the `if` conditional what the field is for.

Reply:
Creating a new type for a boolean seems completely overkill. The argument name is pretty explicit in itself I think. Here what I will do: create another function called `document_full_collapsible` so that it doesn't require an extra argument and it will be explicit what it's doing. :)

Reply:
The beauty of Rust's enums is that they're really easy to make and use. :-) But I'm good with `document_full_collapsible` if @camelid is.

Another possibility, going back to my idea of writing out the `<details>` wrapper at the call site: What if `document_full` returns an `Option<Buffer>`? Then the call sites can either splat the contents directly into their copy of `w`, or wrap them in `<details>` as appropriate.

Reply:
I didn't want to create a new `Buffer`, but that could work too, just not as performant.